tools/wpt/android.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import os
+import platform
+import shutil
+import subprocess
+
+import requests
+from mozrunner.devices import android_device
+
+android_device.TOOLTOOL_PATH = os.path.join(os.path.dirname(__file__),
+                                            os.pardir,
+                                            "third_party",
+                                            "tooltool",
+                                            "tooltool.py")
+
+here = os.path.abspath(os.path.dirname(__file__))
+wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
+
+
+def get_parser_install():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--reinstall", action="store_true", default=False,
+                        help="Force reinstall even if the emulator already exists")
+    return parser
+
+
+def get_parser_start():
+    return get_parser_install()
+
+
+def get_sdk_path(dest):
+    if dest is None:
+        # os.getcwd() doesn't include the venv path
+        dest = os.path.join(wpt_root, "_venv")
+    dest = os.path.join(dest, 'android-sdk')
+    return os.path.abspath(os.environ.get('ANDROID_SDK_PATH', dest))
+
+
+def uninstall_sdk(dest=None):
+    path = get_sdk_path(dest)
+    if os.path.exists(path) and os.path.isdir(path):
+        shutil.rmtree(path)
+
+
+def install_sdk(logger, dest=None):
+    sdk_path = get_sdk_path(dest)
+    if os.path.isdir(sdk_path):
+        logger.info("Using SDK installed at %s" % sdk_path)
+        return sdk_path, False
+
+    if not os.path.exists(sdk_path):
+        os.makedirs(sdk_path)
+
+    os_name = platform.system().lower()
+    if os_name not in ["darwin", "linux", "windows"]:
+        logger.error("Unsupported platform %s" % os_name)
+        raise NotImplementedError
+
+    os_name = 'darwin' if os_name == 'macosx' else os_name
+    url = 'https://dl.google.com/android/repository/sdk-tools-%s-4333796.zip' % (os_name,)
+
+    logger.info("Getting SDK from %s" % url)
+    temp_path = os.path.join(sdk_path, url.rsplit("/", 1)[1])
+    try:
+        with open(temp_path, "wb") as f:
+            with requests.get(url, stream=True) as resp:
+                shutil.copyfileobj(resp.raw, f)
+
+        # Python's zipfile module doesn't seem to work here
+        subprocess.check_call(["unzip", temp_path], cwd=sdk_path)
+    finally:
+        os.unlink(temp_path)
+
+    return sdk_path, True
+
+
+def install_android_packages(logger, sdk_path, no_prompt=False):
+    sdk_manager_path = os.path.join(sdk_path, "tools", "bin", "sdkmanager")
+    if not os.path.exists(sdk_manager_path):
+        raise OSError("Can't find sdkmanager at %s" % sdk_manager_path)
+
+    #TODO: Not sure what's really needed here
```

Comment:
Looks right. https://hg.mozilla.org/mozilla-central/file/tip/python/mozboot/mozboot/android-packages.txt has newer versions.

I assume you're generally following https://hg.mozilla.org/mozilla-central/file/tip/python/mozboot/mozboot/android.py

Reply:
Yeah, mostly following that.